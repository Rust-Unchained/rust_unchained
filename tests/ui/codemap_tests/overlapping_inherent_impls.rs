// Test that you cannot define items with the same name in overlapping inherent
// impl blocks.

#![allow(unused)]

struct Foo;

impl Foo {
    fn id() {} //~ ERROR duplicate definitions
}

impl Foo {
    fn id() {}
}

struct Bar<T>(T);

impl<T> Bar<T> {
    fn bar(&self) {} //~ ERROR duplicate definitions
}

impl Bar<u32> {
    fn bar(&self) {}
}

struct Baz<T>(T);

impl<T: Copy> Baz<T> {
	// Allowed with Rust Unchained, Vec cannot implement Copy so these impls don't overlap
    fn baz(&self) {}
}

impl<T> Baz<Vec<T>> {
    fn baz(&self) {}
}

impl<T: Clone> Baz<T> {
    fn foo(&self) {} //~ ERROR duplicate definitions
}

impl<T> Baz<Vec<T>> {
    fn foo(&self) {}
}

fn main() {}
