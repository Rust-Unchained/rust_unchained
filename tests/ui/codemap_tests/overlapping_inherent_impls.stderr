error[E0592]: duplicate definitions with name `id`
  --> $DIR/overlapping_inherent_impls.rs:9:5
   |
LL |     fn id() {}
   |     ^^^^^^^ duplicate definitions for `id`
...
LL |     fn id() {}
   |     ------- other definition for `id`

error[E0592]: duplicate definitions with name `bar`
  --> $DIR/overlapping_inherent_impls.rs:19:5
   |
LL |     fn bar(&self) {}
   |     ^^^^^^^^^^^^^ duplicate definitions for `bar`
...
LL |     fn bar(&self) {}
   |     ------------- other definition for `bar`

error[E0592]: duplicate definitions with name `foo`
  --> $DIR/overlapping_inherent_impls.rs:38:5
   |
LL |     fn foo(&self) {}
   |     ^^^^^^^^^^^^^ duplicate definitions for `foo`
...
LL |     fn foo(&self) {}
   |     ------------- other definition for `foo`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0592`.
